packages/fx-core/resource/yaml-schema.json

```diff
@@ -0,0 +1,1495 @@
+      "description": "projectId used for telemetry"
+    },
+    "environmentFolderPath": {
+      "type": "string",
+      "description": "path to the environment folder"
+    },
+    "version": {
+      "type": "string",
+      "description": "The version of the yaml file schema"
+    },
+    "provision": {
+      "$ref": "#/definitions/lifeCycleArray",
+      "description": "Triggered by `teamsfx provision`"
+    },
+    "deploy": {
+      "$ref": "#/definitions/lifeCycleArray",
+      "description": "Triggered by `teamsfx deploy`"
+    },
+    "publish": {
+      "$ref": "#/definitions/lifeCycleArray",
+      "description": "Triggered by `teamsfx publish`"
+    }
+  },
+  "required": ["version"],
+  "definitions": {
+    "lifeCycleArray": {
+      "type": "array",
+      "items": {
+        "anyOf": [
+          { "$ref": "#/definitions/aadAppCreate" },
+          { "$ref": "#/definitions/aadAppUpdate" },
+          { "$ref": "#/definitions/armDeploy" },
+          { "$ref": "#/definitions/azureStorageEnableStaticWebsite" },
+          { "$ref": "#/definitions/cliRunNpmCommand" },
+          { "$ref": "#/definitions/cliRunDotnetCommand" },
+          { "$ref": "#/definitions/cliRunNpxCommand" },
+          { "$ref": "#/definitions/azureStorageDeploy" },
+          { "$ref": "#/definitions/azureAppServiceDeploy" },
+          { "$ref": "#/definitions/azureFunctionsDeploy" },
+          { "$ref": "#/definitions/teamsAppCreate" },
+          { "$ref": "#/definitions/teamsAppValidateManifest" },
+          { "$ref": "#/definitions/teamsAppValidateAppPackage" },
+          { "$ref": "#/definitions/teamsAppZipAppPackage" },
+          { "$ref": "#/definitions/teamsAppUpdate" },
+          { "$ref": "#/definitions/teamsAppPublishAppPackage" },
+          { "$ref": "#/definitions/botAadAppCreate" },
+          { "$ref": "#/definitions/botframeworkCreate" },
+          { "$ref": "#/definitions/fileCreateOrUpdateEnvironmentFile" },
+          { "$ref": "#/definitions/fileCreateOrUpdateJsonFile" },
+          { "$ref": "#/definitions/fileUpdateEnv" },
+          { "$ref": "#/definitions/fileUpdateJson" },
+          { "$ref": "#/definitions/prerequisiteInstall" },
+          { "$ref": "#/definitions/m365TitleAcquire" },
+          { "$ref": "#/definitions/spfxDeploy" },
+          { "$ref": "#/definitions/teamsAppCopyAppPackageToSPFx" },
+          { "$ref": "#/definitions/script" }
+        ]
+      }
+    },
+    "aadAppCreateBase": {
+      "type": "object",
+      "description": "Create Azure Active Directory application and client secret (optional). Please refer to https://aka.ms/teamsfx-actions/aadapp-create for more details.",
+      "required": ["uses", "writeToEnvironmentFile"],
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "An optional name of this action"
+        },
+        "env": {
+          "type": "object",
+          "description": "Define environment variables",
```

Comment:
What is this? Is this inputs or outputs?

Reply:
It's inputs. It's for defining environemnt variables in the context of this action, which is especially useful for `cli/npmRunScript`, `script` and actions for deployment.